### DISCUSSION TITLE
Cloudwatch async calls on Ubuntu 22.04

### DISCUSSION THREAD

**Discussion by disa6302** [aws/aws-sdk-cpp#2671] [Q&A] [answered]
Hello,

I have been using AWS CPP SDK 1.11.92 for quite some time now and everything was working fine. My application uses cloudwatch  monitoring to push application performance metrics. I was using the `PutMetricDataAsync` that is invoked from different threads and callbacks when a metric needs to be pushed. This was working perfectly fine on ubuntu 18.04, but the moment I switched to 22.04, I saw PutMetricDataAsync backing up and not clearing up at all which means no metrics are actually getting pushed to cloudwatch. I am using system Openssl and curl. Openssl version is 3.0.0. The cmake command I am running is:

`cmake .. -DCUSTOM_MEMORY_MANAGEMENT=OFF -DENABLE_TESTING -DBUILD_ONLY="monitoring;logs"`

Has anything changed in the way this API is supposed to be used and in the cleanup process? Is there a sample application available for me to reference?

**Comment by jmklix** [comment, marked as answer]
I don't think anything has changed recently. You can find all of the samples [here](https://github.com/awsdocs/aws-doc-sdk-examples/tree/main/cpp/example_code) but I don't think there will be a sample with exactly what you are looking for. Can you confirm that it still works on ubuntu 18.04 and if it does pasta a minimal reproduction sample that shows the same behavior you are seeing on 22.04?

**Reply by jmklix** [reply]
Please let us know if you are having any problems with any updates. Marking this as answered for now

**Comment by lucas-0liveira** [comment]
Hello! It seems like you're working with the cmake command and configuring some options. As for the API, without more context about which API you're using, I can't provide specific information about potential changes or updates.

Regarding the cleaning process, there usually aren't many changes in using cmake for project cleanup. The common command is make clean or cmake --build . --target clean, but this can vary depending on the project's structure and specific settings.

**Reply by kellertk** [reply]
Hi there. Please excuse me if I'm not correct, but it looks like this answer is copy/pasted from an LLM tool. Thank you for trying to help people, but simply copy/paste'ing from an AI tool is not helpful. You can tell which API @disa6302 is using by reading the question - they are talking about the AWS CPP SDK, after updating from version 1.11.92. We ask that if you choose to answer questions here, please respect everyone's time by making sure that your answers are relevant to the question asked.

**Reply by lucas-0liveira** [reply]
I am not using AI. If you interpreted my response as copy/paste, I'm sorry, it was my own opinion. Here on GitHub, I learn just as much when giving an answer as when someone asks and discusses a specific subject. I believe you could simply say that my answer wasn't 'correct' and then proceed to provide the solution. I'm here to both help and learn more about the subject!
